google_analytics/aggregation.py

```diff
@@ -1,7 +1,138 @@
+import json
 import os
+import shutil
+import glob
+import re
 import warnings
-import numpy as np
+
 import pandas as pd
+import numpy as np
+
+
+def reduce_df(path, output, nrows=None, chunksize=20000):
+    """ Load Google analytics data from JSON into a Pandas.DataFrame. """
+    if nrows and chunksize:
+        msg = "Reading {} rows in chunks of {}. We are gonna need {} chunks"
+        print(msg.format(nrows, chunksize, nrows / chunksize))
+
+    temp_dir = "../data/temp"
```

Comment:
We are not 'cleaning' this directory. So right now we are loading the train set in 86 chucks and saving them in the folder temp. Then we load everything in this folder and merge it. Here everything goes fine. Then we are loading the test set in 21 chucks and saving them in the same temp folder. Basically, we are writing over the first 21 chucks of the train set. Then we are concatenating everything in the folder temp, which includes both the test set (chunk 0 till 20) and the last part of the train set (chunck 21 till 86). 
So basically, we need to empty the temp folder before starting with a new dataset.

Reply:
This should do the job:     
for the_file in os.listdir(temp_dir):
        try:
            os.unlink(os.path.join(temp_dir, the_file))
        except Exception as e:
            print(e)

Reply:
thats a huge bug that I didnt notice thanks!